src/main/webapp/app/shared/user-settings/personal-access-tokens/personal-access-tokens.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { FeatureToggle } from 'app/shared/feature-toggle/feature-toggle.service';
+import { PersonalAccessTokenService } from 'app/shared/user-settings/personal-access-tokens/personal-access-tokens.service';
+
+@Component({
+    selector: 'jhi-personal-access-tokens',
+    templateUrl: './personal-access-tokens.component.html',
+    styleUrls: ['../user-settings.scss'],
+})
+export class PersonalAccessTokensComponent implements OnInit {
+    FeatureToggle = FeatureToggle;
+    tokenMaxLifetimeDays: number;
+    newTokenLifetimeDays = 1;
+    newTokenLifetimeMillis: number;
+    wasCopied = false;
+    token = '';
+
+    private dayInMillis = 1000 * 60 * 60 * 24;
+
+    constructor(private patService: PersonalAccessTokenService) {
+        this.convertTokenLifetimeFromDaysToMillis();
+    }
+
+    ngOnInit() {
+        this.patService.getTokenMaxLifetimeMillis().subscribe((resp: number) => (this.tokenMaxLifetimeDays = resp / this.dayInMillis));
```

Comment:
`resp / this.dayInMillis` is probably not an integer-division in JavaScript, but auto-converts to floats which results in partial days. Should there be some rounding down? I suppose rounding here and restricting the input field to an integer number of days is enough for users.

Reply:
What is the common way to limit input fields to interges? Using regex?

Reply:
I think adding a `step="1"` to the `<input …>` tag might be enough.

Reply:
`[step]="1"` should be the same, correct? I'm a bit unsure because it's front end development :)

Reply:
`[step]="1"` is an Angular binding as marked by the brackets. So instead of `1` you could have any JavaScript expression so that you can dynamically change the step size via attributes of the component class.
We don’t need that here, so I think `step="1"` is preferable, even though it should work the same.

Reply:
The same then also applies to `[customMin]="1"` instead of `min="1"`.

Reply:
Thanks for clarification. I'm not an expert regarding angular :) I'll fix that.

Reply:
I've replaced the the parts.

Reply:
> ~The same then also applies to `[customMin]="1"` instead of `min="1"`.~
> 
> Edit: That is probably part of some validation directive and should remain as-is.

Ok. I'll revert that :)